src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -136,9 +136,13 @@ public class PipelineCacheServer
                     
                     if (!string.IsNullOrEmpty(variableToSetOnHit))
                     {
-                        context.SetVariable($"{PipelineArtifactConstants.PipelineCache}.{variableToSetOnHit}", "True");
+                        if(context.Variables.ContainsKey(variableToSetOnHit))
+                        {
+                            context.Output($"Overwriting {variableToSetOnHit}");
+                        }
+                        context.SetVariable($"{variableToSetOnHit}", "True");
```

Comment:
`true` (all lowercase) would be better here. It's more universally accepted and parsed/evaluated as a boolean. `True` (title case) was historically a .NET thing, although .NET now ignores case when converting a string to a boolean.

@chrispat - thoughts?

Btw, we should **always** set (or unset) this variable. If `cacheHitVar` had a value prior to the step, a cache miss should unset/clear its value (or set it to `false`). I don't think we should hold up the PR for this right now, but should be addressed at some point.

Reply:
+1 to change to "true"; "True" is also in Python 

---
In reply to: [299487780](https://github.com/microsoft/azure-pipelines-agent/pull/2320#discussion_r299487780) [](ancestors = 299487780)

Reply:
I pushed a change for this. Now we set 'true' or 'false' based on whether the cache was restored or not.